examples/uploader/uploader

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# Copyright 2021-2022 python-tuf contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Simple uploader tool example
+
+Uploader is a maintainer application that communicates with the repository
+example. Uploader controls offline signing keys and produces signed metadata
+that it sends to the repository application so that the metadata can be added
+to the repository.
+"""
+
+import argparse
+import logging
+import os
+import sys
+from hashlib import sha256
+from pathlib import Path
+from typing import List, Optional
+from urllib import request
+
+from _localrepo import LocalRepository
+
+logger = logging.getLogger(__name__)
+
+
+def build_metadata_dir(base_url: str) -> str:
+    """build a unique and reproducible metadata dirname for the repo url"""
+    name = sha256(base_url.encode()).hexdigest()[:8]
```

Comment:
I used `urllib.parse.quote(url, safe="")` in sigstore-python: The directory names are a bit long (this becomes an issue on some filesystems) but otherwise it's more debuggable than the hash used here

Reply:
eh, this matches what client example does so I'll just resolve this without changes